front/src/js/paint/structDate/color.ts

```diff
@@ -158,33 +140,27 @@ export class Color {
 
         switch (hi) {
             case 0:
-                rgb = new RGB(v, vInc, vMin);
-                break;
+                return new RGB(v, vInc, vMin);
 
             case 1:
-                rgb = new RGB(vDec, v, vMin);
-                break;
+                return new RGB(vDec, v, vMin);
 
             case 2:
-                rgb = new RGB(vMin, v, vInc);
-                break;
+                return new RGB(vMin, v, vInc);
 
             case 3:
-                rgb = new RGB(vMin, vDec, v);
-                break;
+                return new RGB(vMin, vDec, v);
 
             case 4:
-                rgb = new RGB(vInc, vMin, v);
-                break;
+                return new RGB(vInc, vMin, v);
 
             case 5:
-                rgb = new RGB(v, vMin, vDec);
-                break;
+                return new RGB(v, vMin, vDec);
         }
-        return rgb;
+        throw new Error("Variable hi has a bad value");
```

Comment:
no need because 
let hi: number = Math.floor(h / 60) % 6;
and hi always int [0 5]